packages/commonwealth/client/scripts/views/pages/view_thread/ThreadPollCard.tsx

```diff
@@ -2,65 +2,123 @@ import { notifyError, notifySuccess } from 'controllers/app/notifications';
+import { openConfirmation } from 'views/modals/confirmation_modal';
 import type Poll from '../../../models/Poll';
+import { PollCard } from '../../components/Polls';
 import { CWModal } from '../../components/component_kit/new_designs/CWModal';
-import { PollCard } from '../../components/poll_card';
 import { OffchainVotingModal } from '../../modals/offchain_voting_modal';
-import { getPollTimestamp, handlePollVote } from './helpers';
+import { getPollTimestamp } from './helpers';
 
 type ThreadPollCardProps = {
   poll: Poll;
-  onVote: () => void;
   showDeleteButton?: boolean;
-  onDelete?: () => void;
   isTopicMembershipRestricted?: boolean;
 };
 
 export const ThreadPollCard = ({
   poll,
-  onVote,
   showDeleteButton,
-  onDelete,
   isTopicMembershipRestricted = false,
 }: ThreadPollCardProps) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
 
   const user = useUserStore();
 
+  const { mutateAsync: deletePoll } = useDeletePollMutation({
+    threadId: poll.threadId,
+  });
+
+  const { mutateAsync: votePoll } = useVotePollMutation({
+    threadId: poll.threadId,
+  });
+
   const getTooltipErrorMessage = () => {
     if (!user.activeAccount)
       return 'Error: You must join this community to vote.';
     if (isTopicMembershipRestricted) return 'Error: Topic is gated.';
     return '';
   };
 
+  const handleDeletePoll = async () => {
+    try {
+      await deletePoll({
+        pollId: poll.id,
+        address: user.activeAccount?.address || '',
+        authorCommunity: user.activeAccount?.community?.id || '',
+      });
+      notifySuccess('Poll deleted');
+    } catch (e) {
+      console.error(e);
+      notifyError('Failed to delete poll');
+    }
+  };
+
+  const handlePollVote = (
+    votedPoll: Poll,
+    option: string,
+    isSelected: boolean,
+  ) => {
+    if (!user.isLoggedIn || !user.activeAccount || isSelected) {
+      return;
+    }
+
+    openConfirmation({
+      title: 'Info',
+      description: `Submit a vote for '${option}'?`,
+      buttons: [
+        {
+          label: 'Submit',
+          buttonType: 'primary',
+          buttonHeight: 'sm',
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+          onClick: async () => {
+            const selectedOption = votedPoll.options.find((o) => o === option);
+
+            if (!selectedOption) {
+              notifyError('Invalid voting option');
+              return;
+            }
+
+            try {
+              await votePoll({
+                pollId: votedPoll.id,
+                communityId: votedPoll.communityId,
+                authorCommunityId: user.activeAccount?.community?.id || '',
+                address: user.activeAccount?.address || '',
+                selectedOption,
+              }).catch(console.error);
+            } catch (err) {
+              console.error(err);
+              notifyError(
+                'Error submitting vote. Maybe the poll has already ended?',
```

Comment:
i think we can update the msg. `?` might make it confusing for user

Reply:
definitely. again, something I didn't touch